src/editors/EditorHeader.jsx

```diff
@@ -1,44 +1,112 @@
 import {
-  ActionRow, IconButton, Icon, ModalDialog,
+  ActionRow, IconButton, Icon, ModalDialog, Form,
 } from '@edx/paragon';
-import PropTypes from 'prop-types';
-import React, { useContext } from 'react';
-import { Close } from '@edx/paragon/icons';
+import React, { useContext, useState, useEffect } from 'react';
+import { Edit, Close } from '@edx/paragon/icons';
 import EditorPageContext from './EditorPageContext';
-import { ActionStates, mapBlockTypeToName } from './data/constants';
+import { ActionStates } from './data/constants';
 
-const EditorHeader = ({ title }) => {
-  const { unitUrl, unitUrlLoading, studioEndpointUrl } = useContext(EditorPageContext);
+/* The Header component renders based on three states:
+    loading -> loaded but not editing -> editing. */
+
+const EditorHeader = () => {
+  const {
+    titleRef, editorRef, setBlockTitle,
+    blockValue, blockLoading,
+    unitUrl, unitUrlLoading, studioEndpointUrl,
+  } = useContext(EditorPageContext);
+
+  const [editing, setEditing] = useState(false);
+
+  const [title, setTitle] = useState('');
+  useEffect(() => {
+    if (blockLoading === ActionStates.FINISHED) {
+      setTitle(blockValue ? blockValue.data.display_name : '');
```

Comment:
Ok so the default title will be '' if the xbock doesn't provide one. Is there a chance we could make this the type of the xblock? Does that make sense?

Reply:
Currently, when entering empty text into the text editor title, it will display "Editing: Component" and it will save as "html" (then when you reopen, it will display "Editing: html"). I think type of xblock would make sense here. Thoughts?